sshd/src/main/scala/ammonite/repl/sshd/ShellSession.scala

```diff
@@ -0,0 +1,63 @@
+package ammonite.repl.sshd
+
+import java.io.{InputStream, OutputStream}
+
+import ammonite.repl.sshd.util.Implicits
+import org.apache.sshd
+
+/**
+ * Implementation of ssh server's remote shell session, which will be serving remote user.
+ * @param remoteShell actual shell implementation, which will serving remote user's shell session.
+ */
+private[sshd] class ShellSession(remoteShell: SshServer.RemoteShellServer) extends sshd.server.Command {
+  var in: InputStream = _
+  var out: OutputStream = _
+  var exit: Option[sshd.server.ExitCallback] = None
+  var thread: Option[Thread] = None
+
+  def setInputStream(in: InputStream) = { this.in = in }
+  def setOutputStream(out: OutputStream) = { this.out = new SshOutputStream(out) }
+  def setErrorStream(err: OutputStream) = { }
+
+  // called by ssh server to instrument this session with a callback that it finished serving a user
+  def setExitCallback(exit: sshd.server.ExitCallback) { this.exit = Option(exit) }
+
+  // called then ssh server is ready to start this session. Starts the actual shell-serving task.
+  def start(env: sshd.server.Environment) {
+    import Implicits._
+    val thread = new Thread({
```

Comment:
I don't think this does what you think it does, due to the way implicits work

```scala
scala> implicit def runnable(block: => Unit) = new Runnable{ override def run() = block }
warning: there was one feature warning; re-run with -feature for details
runnable: (block: => Unit)Runnable

scala> {println(1); println(2)}: Runnable
1
res0: Runnable = $anon$1@234bef66

scala> res0.run()
2
```

This one really shouldn't be an implicit, and since it's called only once it probably shouldn't even be a helper